src/main/java/edu/sfsu/HtmlFormatter.java

```diff
@@ -110,6 +113,32 @@ public static String generateHtml(Student student) {
 
         fragment.p("comments", student.comment);
 
+        generateCheckpoint(fragment, student.id, student.checkpointOralPresentation,
```

Comment:
Why do you need to duplicate these lines? A little further below there is exactly the same statement.